Ct1419 feature switch

stephenrichards opened the pull request.

EdwardAndress commented:
- The frozen literals cop is intended to ease transition to ruby 3.0
- I've disabled it here but we should revisit this later

aliuk2012 commented on config/settings.yml:
```diff
@@ -0,0 +1,10 @@
+  - courts
+  - legal_aid
+  - probation
+  - i_don't_know/more_than_one/other
```
Should this maybe be stored in the DB and exposed via an API? I'm just thinking that we will have this list defined in two places (once in public and once in staff).

stephenrichards committed:
CT-1419 FeatureSets

This PR provdes the ability to set configuration for enabled features on each environment
and then interrogate them.  For example, to set a feature 'sars' enabled on localhost in
rails environments development and test, and on the demo server -  but disabled elsewhere,
write the following in the config/settings.yml file:

enabled_features:
  sars:
    Rails-development: true
    Host-dev: false
    Host-demo: true
    Host-staging: false
    Host-production: false

Then in the code, you can interrogate whether or not it is enabled by:

   FeatureSet.sars.enabled?

The feature set can be enabled/disabled on the fly with:

   FeatureSet.sars.disable!
   FeatureSet.sars.enable!

stephenrichards committed:
Add respond_to_missing? and change HostEnv to Local for development env

stephenrichards committed:
Update feature set settings file

change setting from Rails-development to Local
